catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -117,96 +125,29 @@ public void setHistorian(Historian historian) {
-
-        validateCreateRequest(createRequest);
-
-        if (Requests.isLocal(createRequest)
-                && !sourceOperations.isSourceAvailable(sourceOperations.getCatalog())) {
-            SourceUnavailableException sourceUnavailableException = new SourceUnavailableException(
-                    "Local provider is not available, cannot perform create operation.");
-            if (INGEST_LOGGER.isWarnEnabled()) {
-                INGEST_LOGGER.warn(
-                        "Error on create operation, local provider not available. {} metacards failed to ingest. {}",
-                        createRequest.getMetacards()
-                                .size(),
-                        buildIngestLog(createRequest),
-                        sourceUnavailableException);
-            }
-            throw sourceUnavailableException;
-        }
-
         CreateResponse createResponse = null;
 
         Exception ingestError = null;
-        try {
-            createRequest = injectAttributes(createRequest);
-
-            setDefaultValues(createRequest);
-
-            Map<String, Serializable> unmodifiablePropertiesMap = Collections.unmodifiableMap(
-                    createRequest.getProperties());
-            HashMap<String, Set<String>> requestPolicyMap = new HashMap<>();
-            for (Metacard metacard : createRequest.getMetacards()) {
-                HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
-                for (PolicyPlugin plugin : frameworkProperties.getPolicyPlugins()) {
-                    PolicyResponse policyResponse = plugin.processPreCreate(metacard,
-                            unmodifiablePropertiesMap);
-                    opsSecuritySupport.buildPolicyMap(itemPolicyMap,
-                            policyResponse.itemPolicy()
-                                    .entrySet());
-                    opsSecuritySupport.buildPolicyMap(requestPolicyMap,
-                            policyResponse.operationPolicy()
-                                    .entrySet());
-                }
 
-                metacard.setAttribute(new AttributeImpl(Metacard.SECURITY, itemPolicyMap));
-            }
-            createRequest.getProperties()
-                    .put(PolicyPlugin.OPERATION_SECURITY, requestPolicyMap);
+        createRequest = queryOperations.setFlagsOnRequest(createRequest);
+        createRequest = validateCreateRequest(createRequest);
+        createRequest = validateLocalSource(createRequest);
 
-            for (AccessPlugin plugin : frameworkProperties.getAccessPlugins()) {
-                createRequest = plugin.processPreCreate(createRequest);
-            }
+        try {
+            createRequest = injectAttributes(createRequest);
+            createRequest = setDefaultValues(createRequest);
+            createRequest = updateCreateRequestPolicyMap(createRequest);
+            createRequest = processPrecreateAccessPlugins(createRequest);
```

Comment:
man there's so much similiarity... i wish there was just someway to clean this up ;)